### ISSUE TITLE
[MNG-5940] Change the maven-source-plugin jar goal into jar-no-fork in Maven Super POM

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7499]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** opened **[MNG-5940](https://issues.apache.org/jira/browse/MNG-5940?redirect=false)** and commented

At the moment the `maven-source-plugin:jar` goal is defined in the Maven super pom:

```xml
<plugin>
          <inherited>true</inherited>
          <artifactId>maven-source-plugin</artifactId>
          <executions>
            <execution>
              <id>attach-sources</id>
              <goals>
                <goal>jar</goal>
              </goals>
            </execution>
          </executions>
        </plugin>
```

where the goal of `maven-source-plugin` should be changed from `jar` into `jar-no-fork`, cause most of the time you need to override this behaviour.

---

**Issue Links:**
- [MNG-5863](https://issues.apache.org/jira/browse/MNG-5863) default pom's release-profile should invoke source plugin with goal "jar-no-fork" instead of "jar"
 (_**"duplicates"**_)
- [MNG-483](https://issues.apache.org/jira/browse/MNG-483) sort out artifact attachment

- [MSOURCES-121](https://issues.apache.org/jira/browse/MSOURCES-121) Check for duplicated addition of the same file

- [MNG-6405](https://issues.apache.org/jira/browse/MNG-6405) Incorrect basedir in forked executions when using flatten-maven-plugin with custom outputDirectory

- [MNG-7029](https://issues.apache.org/jira/browse/MNG-7029) Remove super POM "release-profile" release profile

**Remote Links:**
- [Page
](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=67636510)
- [GitHub Pull Request #168
](https://github.com/apache/maven/pull/168)

1 votes, 11 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Strongly support this one.

**Comment by jira-importer** [comment]
**[Daniel Roig](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=droig)** commented

This will be a much-awaited fix for my part. It manifested itself quite strangely. In Maven 3.3.9 it made the `source` plugin deploy twice, which Artifactory didn't think was too amusing when performing a release.

If I used (in 3.3.9):

```
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-source-plugin</artifactId>
	<executions>
		<execution>
			<id>attach-sources</id>
			<phase>package</phase>
			<goals>
				<goal>jar-no-fork</goal>
				<goal>test-jar-no-fork</goal>
			</goals>
		</execution>
	</executions>
</plugin>
```

I would wind up with the effective pom:

```
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-source-plugin</artifactId>
	<executions>
		<execution>
			<id>attach-sources</id>
			<phase>package</phase>
			<goals>
                                <goal>jar</goal>
				<goal>jar-no-fork</goal>
				<goal>test-jar-no-fork</goal>
			</goals>
		</execution>
	</executions>
</plugin>
```

But it somehow worked fine in Maven 3.0.5. Now, in 3.3.9, I use:

```
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-source-plugin</artifactId>
	<executions>
		<execution>
			<id>attach-sources</id>
			<phase>package</phase>
			<goals>
                                <goal>jar</goal>
				<goal>test-jar-no-fork</goal>
			</goals>
		</execution>
	</executions>
</plugin>
```

in order to match the super POM.

Nevertheless, thanks for fixing this. I'll be sure to remember to change my POM back to `<goal>jar-no-fork</goal>` when 3.4.0 is released.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

[Daniel Roig](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=droig) Hi Daniel, you can workaround that via [this](http://blog.soebes.de/blog/2016/07/01/spoiler-alert-maven-3-dot-4-0-changes-ii/).

**Comment by jira-importer** [comment]
**[Christian Schulte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schulte77)** commented

This has been reverted in 3.4 and postponed to model version 5.0.0.

**Comment by jira-importer** [comment]
**[Christian Schulte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schulte77)** commented

Part of 3.4. See [this mail](http://mail-archives.apache.org/mod_mbox/maven-dev/201610.mbox/%3CCA%2BnPnMwwC7gpBe3u%2BsK8dPs5St3trAsgMXwUWvL7ksrj4tmqSQ%40mail.gmail.com%3E).

**Comment by jira-importer** [comment]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** commented

Maven 3.4.0 has been dropped. See [this thread](https://mail-archives.apache.org/mod_mbox/maven-dev/201701.mbox/%3CCA%2BnPnMx-e7kGYy3Hp87v8hLGdhp1q%3DtKLx_6QuZ4kGUqHEBGcw%40mail.gmail.com%3E) for more details.

This issue will need to be re-scheduled for a Maven release in the (hopefully near) future.

**Comment by jira-importer** [comment]
**[Matt Nelson](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mattnelson)** commented

Could this issue be retargeted to 3.5.1?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise), the release profile is deprecated anyway. Why change it at all?

**Comment by jira-importer** [comment]
**[Florian Brunner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=puce)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o): the release-profile should either be removed or this issue should be fixed.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Florian Brunner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=puce), makes sense. Branches deployed.

**Comment by jira-importer** [comment]
**[Florian Brunner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=puce)** commented

From the failed build log:

> Caused by: org.codehaus.plexus.resource.loader.ResourceNotFoundException: Could not find resource 'https://www.apache.org/licenses/LICENSE-2.0.txt'.

Do you have this error in other buids, too? Seems unrelated to the change...

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [5b4e173f7d7e84ebbe0a34de6f2a79559e490eb5](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=5b4e173f7d7e84ebbe0a34de6f2a79559e490eb5).
and [dc40536effb28dea0bb481850f29dc22a8e49549](https://gitbox.apache.org/repos/asf?p=maven-site.git;a=commit;h=dc40536effb28dea0bb481850f29dc22a8e49549).
